src/plugins/GUITestBase/src/tests/common_scenarios/msa_editor/multiline/GTTestsMSAMultiline.cpp

```diff
@@ -0,0 +1,1343 @@
+            textItems->append(action->text());
+        }
+    }
+}
+}
+
+GUI_TEST_CLASS_DEFINITION(general_test_0001)
+{
+    // UGENE-7042
+
+    // 1. Open file data/samples/CLUSTALW/COI.aln
+    GTFileDialog::openFile(os, dataDir + "samples/CLUSTALW", "COI.aln");
+    GTUtilsTaskTreeView::waitTaskFinished(os);
+
+    // 2. Switch to multiline mode
+    // Press "Multiline View" button on toolbar
+    GTUtilsMsaEditor::setMultilineMode(os, true);
+
+    // 3. Find seq area #1
+    auto w = GTUtilsMSAEditorSequenceArea::getSequenceArea(os, 1);
+    CHECK_SET_ERR(w != nullptr, QString("Can't find sequence area #1"));
+
+    // 4. Switch to singleline mode
+    // Press "Multiline View" button on toolbar
+    GTUtilsMsaEditor::setMultilineMode(os, false);
+
+    // 5. Find seq area #0, but not #1
+    w = GTUtilsMSAEditorSequenceArea::getSequenceArea(os, 0);
+    CHECK_SET_ERR(w != nullptr, QString("Can't find sequence area #0"));
+    w = GTUtilsMSAEditorSequenceArea::getSequenceArea(os, 1);
+    CHECK_SET_ERR(w == nullptr, QString("Unexpectedly found sequence area #1"));
+
+    GTUtilsMsaEditor::setMultilineMode(os, false);
+}
+
+GUI_TEST_CLASS_DEFINITION(zoom_to_selection_test_0001)
+{
+    // UGENE-7605
+
+    const QString seqName = "Phaneroptera_falcata";
+
+    // Open file data/samples/CLUSTALW/COI.aln
+    GTFileDialog::openFile(os, dataDir + "samples/CLUSTALW", "COI.aln");
+    GTUtilsTaskTreeView::waitTaskFinished(os);
+
+    QAbstractButton* reset_zoom = GTAction::button(os, "Reset Zoom");
+    GTWidget::click(os, reset_zoom);
+    GTUtilsTaskTreeView::waitTaskFinished(os);
+
+    // Switch to multiline mode
+    // Press "Multiline View" button on toolbar
+    GTUtilsMsaEditor::setMultilineMode(os, true);
+
+    // Select seq.
+    GTUtilsMsaEditor::selectRowsByName(os, {seqName});
+
+    reset_zoom = GTAction::button(os, "Reset Zoom");
+    QAbstractButton* zoom_to_sel = GTAction::button(os, "Zoom To Selection");
+    GTWidget::click(os, zoom_to_sel);
+    GTUtilsTaskTreeView::waitTaskFinished(os);
+    GTWidget::click(os, reset_zoom);
+    GTUtilsTaskTreeView::waitTaskFinished(os);
+    GTWidget::click(os, zoom_to_sel);
+    GTUtilsTaskTreeView::waitTaskFinished(os);
+    GTWidget::click(os, reset_zoom);
+    GTUtilsTaskTreeView::waitTaskFinished(os);
+
+    GTUtilsMsaEditor::setMultilineMode(os, false);
+
+    // Must not crash
+}
+
+GUI_TEST_CLASS_DEFINITION(vscroll_test_0001)
+{
+    // UGENE-7522
+
+    // Open file data/samples/CLUSTALW/COI.aln
+    GTFileDialog::openFile(os, dataDir + "samples/CLUSTALW", "COI.aln");
+    GTUtilsTaskTreeView::waitTaskFinished(os);
+
+    // Switch to multiline mode
+    // Press "Multiline View" button on toolbar
+    GTUtilsMsaEditor::setMultilineMode(os, true);
+
+    // Find seq last area
+    int lastWgtIndex = 0;
+    MaEditorWgt *lastWgt = nullptr;
+    for (int i = 0; i < 30; i++) {
+        MaEditorWgt* w = GTUtilsMsaEditor::getEditor(os)->getUI()->getUI(i);
```

Comment:
nit: looks like a good candidate for a helper method in the UI class

Reply:
some time later